{
    "schema_version": "1.6.0",
    "id": "OESA-2024-2254",
    "modified": "2024-10-18T00:00:00Z",
    "published": "2024-10-18T00:00:00Z",
    "withdrawn": "",
    "aliases": null,
    "related": [
        "CVE-2024-42367"
    ],
    "summary": "python-aiohttp security update",
    "details": "Async http client/server framework (asyncio).\r\n\r\nSecurity Fix(es):\r\n\r\naiohttp is an asynchronous HTTP client/server framework for asyncio and Python. Prior to version 3.10.2, static routes which contain files with compressed variants (`.gz` or `.br` extension) are vulnerable to path traversal outside the root directory if those variants are symbolic links. The server protects static routes from path traversal outside the root directory when `follow_symlinks=False` (default).  It does this by resolving the requested URL to an absolute path and then checking that path relative to the root. However, these checks are not performed when looking for compressed variants in the `FileResponse` class, and symbolic links are then automatically followed when performing the `Path.stat()` and `Path.open()` to send the file. Version 3.10.2 contains a patch for the issue.(CVE-2024-42367)",
    "severity": null,
    "affected": [
        {
            "package": {
                "ecosystem": "openEuler-24.03-LTS",
                "name": "python-aiohttp",
                "purl": "pkg:rpm/python-aiohttp@3.9.3-5.oe2403"
            },
            "severity": null,
            "ranges": [
                {
                    "type": "ECOSYSTEM",
                    "repo": "",
                    "events": [
                        {
                            "introduced": "0",
                            "fixed": "3.9.3-5.oe2403",
                            "last_affected": "",
                            "limit": ""
                        }
                    ],
                    "database_specific": null
                }
            ],
            "versions": null,
            "ecosystem_specific": {
                "aarch64": [
                    "python-aiohttp-debuginfo-3.9.3-5.oe2403.aarch64.rpm",
                    "python-aiohttp-debugsource-3.9.3-5.oe2403.aarch64.rpm",
                    "python-aiohttp-help-3.9.3-5.oe2403.aarch64.rpm",
                    "python3-aiohttp-3.9.3-5.oe2403.aarch64.rpm"
                ],
                "src": [
                    "python-aiohttp-3.9.3-5.oe2403.src.rpm"
                ],
                "x86_64": [
                    "python-aiohttp-debuginfo-3.9.3-5.oe2403.x86_64.rpm",
                    "python-aiohttp-debugsource-3.9.3-5.oe2403.x86_64.rpm",
                    "python-aiohttp-help-3.9.3-5.oe2403.x86_64.rpm",
                    "python3-aiohttp-3.9.3-5.oe2403.x86_64.rpm"
                ]
            },
            "database_specific": null
        }
    ],
    "references": [
        {
            "type": "ADVISORY",
            "url": "https://nvd.nist.gov/vuln/detail/CVE-2024-42367"
        }
    ],
    "credits": null,
    "database_specific": {
        "severity": "Medium"
    }
}